apps/glusterfs/volume_entry.go

```diff
@@ -420,11 +420,35 @@ func (v *VolumeEntry) cleanupCreateVolume(db wdb.DB,
 
 	// from a quick read its "safe" to unconditionally try to delete
 	// bricks. TODO: find out if that is true with functional tests
-	DestroyBricks(db, executor, brick_entries)
+	reclaimed, _ := DestroyBricks(db, executor, brick_entries)
+
 	return db.Update(func(tx *bolt.Tx) error {
 		for _, brick := range brick_entries {
 			v.removeBrickFromDb(tx, brick)
 		}
+		// update the device' free/used space after removing bricks
+		for _, b := range brick_entries {
+			for devID, reclaimed := range reclaimed {
```

Comment:
The `reclaimed` type is a map. Unpacking this in a loop is less efficient and more verbose than the alternative of doing map look-ups of the `b.Info.DeviceId` as a key.

Reply:
yeah, this will be more efficient. updated the code.